test/dict.jl

```diff
@@ -228,26 +228,44 @@ end
         d2[k] = v
     end
 
-    @test  isequal(d1, d2)
+    @test eq(d1, d2)
     d3 = copy(d2)
     d4 = copy(d2)
     # Removing an item gives different dict
     delete!(d1, data_in[rand(1:length(data_in))][1])
-    @test !isequal(d1, d2)
+    @test !eq(d1, d2)
     # Changing a value gives different dict
     d3[data_in[rand(1:length(data_in))][1]] = randstring(3)
-    !isequal(d1, d3)
+    !eq(d1, d3)
     # Adding a pair gives different dict
     d4[1001] = randstring(3)
-    @test !isequal(d1, d4)
+    @test !eq(d1, d4)
 
-    @test isequal(Dict(), sizehint!(Dict(),96))
+    @test eq(Dict(), sizehint!(Dict(),96))
 
-    # Here is what currently happens when dictionaries of different types
```

Comment:
I found this comment particularly funny. Having these in 1.0 could be embarrassing... :-)